testsuite/features/step_definitions/common_steps.rb

```diff
@@ -511,8 +511,9 @@
 
 Then(/^I should see the "(.*?)" selected$/) do |product|
   xpath = "//span[contains(text(), '#{product}')]/ancestor::div[contains(@class, 'product-details-wrapper')]"
-  product_identifier = find(:xpath, xpath)['data-identifier']
-  raise "#{product_identifier} is not checked" unless has_checked_field?('checkbox-for-' + product_identifier)
+  product = find(:xpath, xpath)
+  selected = product.find(:xpath, "./div/input[@type='checkbox']").checked?
+  raise "#{product['data-identifier']} is not checked" unless selected
```

Comment:
```suggestion
within(xpath) do
  raise "#{product['data-identifier']} is not checked" unless product.find(:xpath, "./div/input[@type='checkbox']").checked?
end
```

Reply:
why not, it's more elegant as you suggest

BUT `product` is a node, what you want in the error message is `product['data-identifier']`, a string

Reply:
Done.

Reply:
@Bischoff it is impossible that product is undefined, we are passing a value as a parameter in the step definition.
With your extra change, we are adding an extra find, anyhow, as it is only called on a raise, fine with this approach.